components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -3134,4 +3138,35 @@ public static boolean isJITProvisionEnhancedFeatureEnabled() {
         return Boolean.parseBoolean(IdentityUtil.
                     getProperty(FrameworkConstants.ENABLE_JIT_PROVISION_ENHANCE_FEATURE));
     }
+
+    /**
+     * Return a filtered list of requested scope claims.
+     *
+     * @param scopes Requested scopes.
+     * @param tenantDomain tenant domain.
+     * @param claimMappings SpClaimMappings.
+     * @throws ClaimManagementException
+     */
+    public static List<ClaimMapping> getFilteredScopeClaims(Set<String> scopes, String tenantDomain,
+                                                            List<ClaimMapping> claimMappings)
+            throws ClaimManagementException {
+
+        OpenIDConnectClaimFilter filter = new OpenIDConnectClaimFilterImpl();
+        List<String> claimListOfScopes = filter.getClaimsFilteredByOIDCScopes(scopes, tenantDomain);
+        ClaimManagerHandler handler = ClaimManagerHandler.getInstance();
+        List<String> claimMappingListOfScopes = new ArrayList<>();
+        for (String claim : claimListOfScopes) {
+            org.wso2.carbon.user.api.ClaimMapping currentMapping = handler.getClaimMapping(
```

Comment:
Seems there are formatting issues. Please check all the other places as well.

Reply:
resolved